src/TestFramework/Coverage/JUnit/TestFileDataAdder.php

```diff
@@ -33,71 +33,63 @@
 
 declare(strict_types=1);
 
-namespace Infection\TestFramework\Coverage\XmlReport;
+namespace Infection\TestFramework\Coverage\JUnit;
 
 use function explode;
 use Infection\AbstractTestFramework\Coverage\CoverageLineData;
-use Infection\TestFramework\Coverage\CoverageFileData;
-use Infection\TestFramework\Coverage\JUnit\TestFileDataProvider;
-use Infection\TestFramework\PhpUnit\Coverage\IndexXmlCoverageParser;
-use function Safe\file_get_contents;
+use Infection\AbstractTestFramework\TestFrameworkAdapter;
+use Infection\TestFramework\Coverage\CoveredFileData;
 
 /**
+ * Adds test execution info to selected covered file data object.
+ *
  * @internal
  * @final
  */
-class PhpUnitXmlCoverageFactory
+class TestFileDataAdder
 {
-    /**
-     * TODO: make this constant private
-     */
-    public const COVERAGE_INDEX_FILE_NAME = 'index.xml';
-
-    private $coverageDir;
-    private $parser;
     private $testFileDataProvider;
 
+    private $adapter;
+
     public function __construct(
-        string $coverageDir,
-        IndexXmlCoverageParser $coverageXmlParser,
-        ?TestFileDataProvider $testFileDataProvider
+        TestFrameworkAdapter $adapter,
+        TestFileDataProvider $testFileDataProvider
     ) {
-        $this->coverageDir = $coverageDir;
-        $this->parser = $coverageXmlParser;
+        $this->adapter = $adapter;
+
         $this->testFileDataProvider = $testFileDataProvider;
     }
 
     /**
-     * @return array<string, CoverageFileData>
+     * @param iterable<CoveredFileData> $coverage
+     *
+     * @return iterable<CoveredFileData>
      */
-    public function createCoverage(): array
+    public function addTestExecutionInfo(iterable $coverage): iterable
     {
-        $coverageIndexFileContent = file_get_contents(
-            $this->coverageDir . '/' . self::COVERAGE_INDEX_FILE_NAME
-        );
-
-        $coverage = $this->parser->parse($coverageIndexFileContent);
-
-        $this->addTestExecutionInfo($coverage);
+        if (!$this->adapter->hasJUnitReport()) {
+            return $coverage;
+        }
 
-        return $coverage;
+        return $this->testExecutionInfoAdder($coverage);
     }
 
     /**
-     * @param CoverageFileData[] $coverage
+     * @param iterable<CoveredFileData> $coverage
+     *
+     * @return iterable<CoveredFileData>
      */
-    private function addTestExecutionInfo(array $coverage): void
+    private function testExecutionInfoAdder(iterable $coverage): iterable
```

Comment:
not sure about the name of the method. IMO, previous version `addTestExecutionInfo()` is more self explained.

The new name does not mean we return or add anything.

Reply:
I'll bring it back.

Reply:
`addTestExecutionInfo` is still there, a public method line 69 above. So we can have this renamed, but not into `addTestExecutionInfo`

Reply:
I guess it can be inlined here though since the name is not that compelling?

Reply:
`addTestExecutionInfo` isn't a generator, where this method is. So, no, can't be inlined, at least without a closure.

Reply:
OK, I made it an explicit generator. It as actually adding generator. Makes sense?